app/models/concerns/thredded/content_moderation_state.rb

```diff
@@ -30,16 +30,16 @@ module ContentModerationState
       end)
     end
 
-    def visible_to_all?
+    def moderation_state_visible_to_all?
       if Thredded.content_visible_while_pending_moderation
         !blocked?
       else
         approved?
       end
     end
 
-    def visible_to_user?(user)
-      visible_to_all? || (!user.thredded_anonymous? && user_id == user.id)
+    def moderation_state_visible_to_user?(user)
+      moderation_state_visible_to_all? || (!user.thredded_anonymous? && user_id == user.id)
```

Comment:
@glebm can't you just do `moderation_state_visible_to_all? || user_id == user.id` because `NullUsers`'s `user_id` is nil, and `user.id` will never be nil